deltachat-ffi/src/providers.rs

```diff
@@ -12,7 +12,7 @@ pub type dc_provider_t = deltachat_provider_database::Provider;
 pub unsafe extern "C" fn dc_provider_new_from_domain(
     domain: *const libc::c_char,
 ) -> *const dc_provider_t {
-    match deltachat_provider_database::get_provider_info(as_str(domain)) {
+    match deltachat_provider_database::get_provider_info(&to_string_lossy(domain)) {
```

Comment:
does the compiler not complain here? I was under the impression that &to_string_lossy doesn't have a long enough lifetime ...

Reply:
no, this works. 

but a few lines below in dc_provider_new_from_email() i had to move the string to a broader scope

Reply:
https://doc.rust-lang.org/reference/expressions.html#temporary-lifetimes
I'm not claiming I 100% understand it all, but I think this is equivalent to the first example so is fine.